### PR TITLE
Move CI to Jenkins

### PR DESCRIPTION
**Pull request by danimtb**
- Added Jenkinsfile with declarative syntax
- Created scripts to handle the environment setup for each platform
- Only testing in Windows and Linux at the moment (we can add macos later).
- Reduced the combination of pythons x conans to avoid having a slow CI.
- Note that the travis and appveyor files are still not removed (will be done later when Jenkins is configured).

Tested the branch here: https://ci.conan.io/blue/organizations/jenkins/Hooks/detail/feature%2Fmove_ci_to_jenkins/38/pipeline
image

Related to #301
closes #258

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,180 @@
+pipeline {
+    agent none
+
+    stages {
+        stage("Windows") {
```
Comment (danimtb) [inline .ci/Jenkinsfile:5]: In windows we don't have to install the python interpreter, so the setup is faster, so they are run first

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                            }
+                        }
+                        stage('Linux - py35 - conandev') {
+                            environment {
+                                TOXENV = 'py35-conandev'
+                            }
+                            steps {
+                                sh '''
+                                    eval "$(pyenv init -)"
+                                    pyenv activate conan
```
Comment (danimtb) [inline .ci/Jenkinsfile:95]: Didn't want to add anything related to the environment at this step (only run tests), but I couldn't figure it out so I had to place this two lines here

**Comment by danimtb** [comment]
Linux builds can be improved as some python versions are already installed in the images (see https://github.com/conan-io/conan/blob/develop2/.ci/jenkins/testsv2.jenkins), but I think it is fine ATM

**Comment by Croydon** [comment]
Will the logs be still publicly visible for anyone?

**Comment by danimtb** [comment]
@Croydon yes, they should be available in the same way the conan-io/conan ones are right now

**Comment by jgsogo** [comment]
Just checking notifications 😉

**Review by jgsogo — APPROVED** [review]
Good!

**Review by uilianries — APPROVED** [review]
I'm not well familiarized with jenkinsfile, but it looks well ported here. As we are using tox, we don't need to add more steps. This PR is a really great addition! I would a status badge on README file if possible, so we can check the master branch easily.

**Comment by uilianries** [comment]
BTW, can we validate it on OSX too? It would be great, we usually don't have problems in Mac, but it would be a good test scenario too.

**Comment by danimtb** [comment]
> BTW, can we validate it on OSX too? It would be great, we usually don't have problems in Mac, but it would be a good test scenario too.

yes, we can add some Macos configurations as well in a next PR as well as the badge in the readme. I will take a look!

**Pull request merged**